### DISCUSSION TITLE
Android TV / ADB - Controlling Bluetooth

### DISCUSSION THREAD

**Discussion by reinuke** [PRProd/HA-Firemote#341] [Q&A]
This might be a too specific question - Xiaomi Mi Box S + ADB + Bluetooth

Is there a way to trigger restarting the BT module?
Read connected BT devices list? Read known devices? 
A way to reset a specific connection? Re-pair , re-connect through ADB ? 

I have "smart speakers" that go to sleep when no audio is being sent through for 5-6min.
I'd idealy like to create an automation in the style of

-check android tv state (if online, idle, paused, ...,)
-check connected BT
- IF NOT 'speaker' & IF STATE turns to Playing THEN RECONNECT  (or.. send IR blaster signal..)

**Comment by PRProd** [comment]
@reinuke ~

This is beyond the scope of Firemote for sure, but I like a good puzzle.  

Here are my initial thoughts: Do you already have a way of sending an IR blaster signal independently from your Mi Box?  If so, can an IR signal sent from that device be used to wake the speakers?  Maybe the answer is as simple as sending the blaster signal out every 4 minutes while your Mi Box is powered on?

**Reply by reinuke** [reply]
Essentially my Mi Box forgets the one speaker after every restart.
The speaker does go into pairing mode but the MiBox doesn't reconnect.

What I'd like from the Firemote is to show if the BT speaker is connected (maybe a blue glow on the icon) , if not then pressing it it would do a re-pairing. 

I know that I can fetch IF ON and PAIRED DEVICES with "adb shell dumpsys bluetooth_manager" command but I've yet to learn how to connect it via adb or fetch status ..
Hopef if you had any insight.. 
I have gone over adb documentation and tried google-fu but have come out dry as of yet

**Reply by PRProd** [reply]
Sending commands is a mostly reliable task. Trying to get responses to those commands using only Home Assistant integrations is something that I haven't had a lot of success with.  The tricky part for me was getting the timing right.  IF the device replied, how long did it take?  Can I trust the response, or is it stale?  Ultimately, the stability of every scenario I worked on was just too weak, so I stopped working on it.

I've also lost my fair share of time trying to find an ADB command to make a device re-connect with its accessories.  For example, [this issue with Amazon Fire devices](https://github.com/PRProd/HA-Firemote/discussions/115#discussioncomment-5970362) is a particularly frustrating one for many users, including me!  Also, in July-2023 user @michael7oliver asked a somewhat similar question in [discussion #160](https://github.com/PRProd/HA-Firemote/discussions/160).  We worked on a solution together that succeeds, but isn't as elegant as I would have preferred.

I sincerely hope that you DO find a workable solution, and I welcome any additional questions you have that I might be able to help answer.  Best of luck!

**Reply by reinuke** [reply]
Ahh.... BTW.... 

The topics you referenced , I had already gone through them.. 
No Cigar. 

BUT I think I did manage to fix the problem by **not allowing** my speakers to Sleep. 
I think that the speakers, over Bluetooth, receive a signal from the Xiaomi Mi Box ... something in the lines of "Going to shutdown" or "....restart" and they just refuse to reconnect until a certain time has passed. 

When I disabled the Auto-Sleep on the Polk Signa S3 speakers the problem seemed to go away. 
I would still like to create an automation series like "If MAC not Bonded, then CONNECT" and tie it in with "If MAC not Bonded AND NOT Connected THEN IR Blast XXXX" and something like that... 
I have only tested it for 20min but it's something... seems to work in the short end..
